### PR TITLE
Bug 1796412: cluster-reader is unable to view machine resources

### PR DESCRIPTION
**Pull request by Danil-Grigorev**
Added cluster-reader permissions to `cluster-autoscaler` cluster-role

### CODE REVIEW

File: install/03_rbac.yaml [hunk 1]
```diff
@@ -3,6 +3,8 @@ apiVersion: rbac.authorization.k8s.io/v1
 kind: ClusterRole
 metadata:
   name: cluster-autoscaler-operator
+  labels:
```
Comment (enxebre) [inline install/03_rbac.yaml:6]: this clusterRole has more permissions than the cluster reader should have.
We need to create one particular clusterRole reader for it, e.g https://github.com/openshift/cloud-credential-operator/blob/aa07b32fc3440d30bc90af636ca9ccd62e70c58a/manifests/00_clusterreader_clusterrole.yaml

File: hack/gen-crd.sh [hunk 2]
```diff
@@ -3,12 +3,12 @@
 set -eu
 
 function annotate_crd() {
-  script='/^metadata:/a\
-\ \ annotations:\
+  script1='/^  annotations:/a\
 \ \ \ \ exclude.release.openshift.io/internal-openshift-hosted: "true"'
```
Comment (enxebre) [inline hack/gen-crd.sh:7]: this does not seem related to this PR, please let's drop it.

File: install/01_clusterautoscaler.crd.yaml [hunk 3]
```diff
@@ -17,7 +17,6 @@ spec:
     - ca
     singular: clusterautoscaler
   scope: Cluster
-  preserveUnknownFields: false
```
Comment (alexander-demichev) [inline install/01_clusterautoscaler.crd.yaml:20]: Is this required?  @enxebre I remember you introduced `preserveUnknownFields: false` intentionally

File: install/02_machineautoscaler.crd.yaml [hunk 4]
```diff
@@ -38,7 +38,6 @@ spec:
     - ma
     singular: machineautoscaler
   scope: Namespaced
-  preserveUnknownFields: false
```
Comment (enxebre) [inline install/02_machineautoscaler.crd.yaml:41]: this does not seem related to this PR, please let's drop it.

File: install/01_clusterautoscaler.crd.yaml (hunk 3 above)
Comment (alexander-demichev) [reply install/01_clusterautoscaler.crd.yaml:20]: I see the comment below now

File: hack/gen-crd.sh (hunk 2 above)
Comment (Danil-Grigorev) [reply hack/gen-crd.sh:7]: @JoelSpeed going to include the autogen fix for crds in another pr, as I agree it is out of scope

File: install/10_cluster_reader_rbac.yaml [hunk 5]
```diff
@@ -0,0 +1,15 @@
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: cluster-autoscaler-operator:cluster-reader
+  labels:
+    rbac.authorization.k8s.io/aggregate-to-cluster-reader: "true"
```
Comment (enxebre) [inline install/10_cluster_reader_rbac.yaml:6]: is there any docs we can refer here for this label?

Comment (Danil-Grigorev) [reply install/10_cluster_reader_rbac.yaml:6]: Under already created aggregation rule this ClusterRole will be combined with `cluster-reader` list of permissions, according to https://kubernetes.io/docs/reference/access-authn-authz/rbac/#aggregated-clusterroles
